src/darwin/Framework/CHIP/MTRDeviceControllerStorageDelegate.h

```diff
@@ -0,0 +1,119 @@
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRDeviceController.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, MTRStorageSecurityLevel) {
+    // Data must be stored in secure (encrypted) storage.
+    MTRStorageSecurityLevelSecure,
+    // Data may be stored in the clear.
+    MTRStorageSecurityLevelNotSecure,
+} MTR_NEWLY_AVAILABLE;
+
+typedef NS_ENUM(NSUInteger, MTRStorageSharingType) {
+    // Data must not be shared at all (just store locally).
+    MTRStorageSharingTypeNotShared,
+
+    // Data may be shared, but only between controllers that have the same node
+    // identity (same fabric, same node ID).  Sharing not required (could be
+    // stored locally).
+    MTRStorageSharingTypeSameIdentityAllowed,
+
+    // Data must be shared, but only between controllers that have the same node
+    // identity (same fabric, same node ID).
+    MTRStorageSharingTypeSameIdentityRequired,
+
+    // Data may be shared, but only between controllers that have the same
+    // access to devices (e.g. controllers that all have the same CATs if ACLs
+    // are being done via CATs).
+    MTRStorageSharingTypeSameACLs,
+
+    // Data may be, but does not have to be, shared across all controllers on a
```

Comment:
Can we make this more explicit? I don't think we should allow for interpretation of what to do with the storage here.

Reply:
OK.  We can certainly be very opinionated here, at the price of later discovering our opinion is wrong and being somewhat stuck.